subiquity/server/controllers/ssh.py

```diff
@@ -59,8 +68,50 @@ async def GET(self) -> SSHData:
             install_server=self.model.install_server,
             allow_pw=self.model.pwauth)
 
-    async def POST(self, data: SSHData):
+    async def POST(self, data: SSHData) -> None:
         self.model.install_server = data.install_server
         self.model.authorized_keys = data.authorized_keys
         self.model.pwauth = data.allow_pw
         await self.configured()
+
+    async def fetch_id_GET(self, user_id: str) -> SSHFetchIdResponse:
+        identities: List[SSHIdentity] = []
+
+        import_command = ('ssh-import-id', '--output', '-', '--', user_id)
+        env = None
+        if self.app.base_model.proxy.proxy:
+            env = os.environ.copy()
+            env["https_proxy"] = self.app.base_model.proxy.proxy
+
+        try:
+            cp = await arun_command(import_command, check=True, env=env)
+        except subprocess.CalledProcessError as e:
+            log.exception("ssh-import-id failed. stderr: %s", e.stderr)
+            return SSHFetchIdResponse(status=SSHFetchIdStatus.IMPORT_ERROR,
+                                      identities=None, error=e.stderr)
+        keys_material: str = cp.stdout.replace('\r', '').strip()
+
+        # ssh-keygen supports multiple keys at once, but it is simpler to
+        # associate each key with its resulting fingerprint if we call
+        # ssh-keygen multiple times.
+        fingerprint_command = ('ssh-keygen', '-l', '-f', '-')
+        for key_material in (mat for mat in keys_material.splitlines() if mat):
+            try:
+                cp = await arun_command(fingerprint_command, check=True,
+                                        input=key_material)
+            except subprocess.CalledProcessError as e:
+                log.exception("ssh-import-id failed. stderr: %s", e.stderr)
+                return SSHFetchIdResponse(
+                        tatus=SSHFetchIdStatus.FINGERPRINT_ERROR,
```

Comment:
tatus? :)

Reply:
:facepalm: Thanks, I will take this as an opportunity to include unit tests..